addons/web_editor/static/src/components/media_dialog/image_selector.js

```diff
@@ -156,7 +156,24 @@ export class ImageSelector extends FileSelector {
                     });
                     resolve();
                 };
-                imageEl.onload = () => {
+                imageEl.onload = async () => {
+                    const imageType = this.props.media && this.props.media.parentElement.dataset["oeType"];
+                    const imageExtension = IMAGE_EXTENSIONS.find(format => url.split('?')[0].endsWith(format));
```

Comment:
I tried to find the mimetype of the image in a better way as this is already almost done in `validateUrl()` but I did not find a way to keep this information in the `ImageSelector` component. Indeed, `validateUrl()` is called in the context of `FileSelectorControlPanel`. As we are in stable, I did not find a better way than this but I maybe miss something here.